pkg/web/routes.go

```diff
@@ -49,7 +49,13 @@ func (s *Server) setupRoutes() (err error) {
 		// CORS configuration allows the front-end to make cross-origin requests
 		cors.New(corsConf),
 
+		// Maintenance mode middleware to return unavailable
 		s.Maintenance(),
+
+		// Web API and UI Enabled middleware
+		s.APIEnabled(),
+
+		s.UIEnabled(),
```

Comment:
Just using middleware to detect if the type of request is an API or UI request, and returning the appropriate status code if one or the other are disabled. If both are enabled, no middleware is added at all.